### PR TITLE
Feat (v0-glossary): Glossary template

### PR DESCRIPTION
**Pull request by jrakibi**
Integrate Tailwind/Next.js Starter Template into BDP Website

Task in this draft PR includes:
- [x]  Integrate Contentlayer for managing content logic
- [x]  Use MDX and display embedded React components within the content
- [x]  Add tags for topics
- [x]  Add SEO metadata, sitemaps, ..
- [x] Display the content of the markdown
- [x] List all topics + Pagination
- [x]  Sync the glossary files from the remote [repo](https://github.com/bitcoin-dev-project/glossary/) source to our local BDP website
- [x] Add interactive components for first Topic P2PK
- [x] update OG page metadata for all BDP pages

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,27 +1,62 @@
 {
-    "compilerOptions": {
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply tsconfig.json]: done, it's formatted now

[the remark this answers is not in the document]
Comment (Extheoisah) [reply tsconfig.json]: Good point @kouloumos We should actually follow this config as opposed to the one in the transcription frontend. I'll create an issue in that repo and synchronize the prettier and ts configs across all projects.

[the remark this answers is not in the document]
Comment (Extheoisah) [reply tsconfig.json]: Even there might be differences due to the individual repo/project requirements, it should be minimal

**Comment by Extheoisah** [comment]
I changed the base branch to staging instead of master branch

**Comment by Extheoisah** [comment]
Is this PR ready for review? FYI, This needs rebase.

**Comment by jrakibi** [comment]
> Is this PR ready for review? FYI, This needs rebase.

I will make some updates and open it for review today

**Review by IgboPharaoh — COMMENTED** [review]
Good job working on this. As this is a large pull request, I have only been able to review the code changes.
 Also across different pages, there are some comments I added relating to:
- Corrections
- Requested changes
- Questions to better understand your line of thought before further comment.

Please do well to attend to them as I test the latest deployment for this pr.

**Review by jrakibi — COMMENTED** [review]
Thanks @IgboPharaoh  for the mega-review. 
I've addressed these feedbacks in my last commits and added a few responses inline where clarification was needed.

File: app/topics/[...slug]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    const mainContent = coreContent(post)
+    const jsonLd = post.structuredData
+    jsonLd["author"] = authorDetails.map((author) => {
+        return {
+            "@type": "Person",
+            name: author.name
+        }
+    })
+
+    const Layout = PostBDP
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply app/topics/[...slug]/page.tsx]: Renamed to 'Topic Layout'. The idea of layouts is that we can have several ways to represent the UI for the Topic page. Currently, we have only one layout, but we can add more later. So, to make it easy for now I removed the related code and will reintroduce it once we have another layout

File: components/bitcoin-topics/markdown-ui/script-visualizer/scriptVisualizer.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                                        <span className="text-xs text-gray-500">
+                                            {step.description}
+                                        </span>
+                                    </span>
+                                </button>
+                            </li>
+                        ))}
+                    </ol>
+
+                    {/* <div>
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply components/bitcoin-topics/markdown-ui/script-visualizer/scriptVisualizer.tsx]: Removed the whole CodeWindow component, we don’t need it anymore

File: components/bitcoin-topics/topic/PrevNextLinks.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+function PageLink({
+    title,
+    href,
+    dir = "next",
+    ...props
+}: Omit<React.ComponentPropsWithoutRef<"div">, "dir" | "title"> & {
+    title: string
+    href: string
+    dir?: "previous" | "next"
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply components/bitcoin-topics/topic/PrevNextLinks.tsx]: Are you referring to the `dir` variable?
If so, I've renamed it to `direction`

**Comment by Extheoisah** [comment]
@jrakibi Please can you rebase this PR?

File: app/topics/[...slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    const authorList = post?.authors || ["default"]
+    const authorDetails = authorList.map((author) => {
+        const authorResults = allAuthors.find((p) => p.slug === author)
+        return coreContent(authorResults as Authors)
+    })
+    const mainContent = coreContent(post)
+    const jsonLd = post.structuredData
+    jsonLd["author"] = authorDetails.map((author) => {
+        return {
+            "@type": "Person",
```
[the remark this answers is not in the document]
Comment (IgboPharaoh) [reply app/topics/[...slug]/page.tsx:94]: Great

File: components/bitcoin-topics/topic/PrevNextLinks.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (IgboPharaoh) [reply components/bitcoin-topics/topic/PrevNextLinks.tsx]: Yes that was what I was referring to

**Comment by jrakibi** [comment]
> @jrakibi Please can you rebase this PR?

Done

**Comment by IgboPharaoh** [comment]
> @Extheoisah @IgboPharaoh
> 
> Rebased the PR

Nice, will review

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Quite a big PR. Great work overall.
I left some comments and nits.

File: tailwind.config.ts [hunk 6]
```diff
@@ -4,7 +4,10 @@ const config: Config = {
     content: [
         "./pages/**/*.{js,ts,jsx,tsx,mdx}",
         "./components/**/*.{js,ts,jsx,tsx,mdx}",
-        "./app/**/*.{js,ts,jsx,tsx,mdx}"
+        "./app/**/*.{js,ts,jsx,tsx,mdx}",
+        "./node_modules/pliny/**/*.js",
+        "./layouts/**/*.{js,ts,tsx}",
+        "./public/bitcoin-topics/**/*.mdx"
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply tailwind.config.ts:10]: Yes, we will sometimes apply Tailwind styles directly within the markdown

File: components.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
+    "$schema": "https://ui.shadcn.com/schema.json",
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply components.json]: I was using some components from Shadcn library, but eneded up removing them. This file is not needed anymore, deleted in [cdd7bb2](https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/79/commits/cdd7bb235e2ff7bd0b61ec02abc5089d31a9eef1)

File: bun.lockb
[the remark this answers is not in the document]
Comment (jrakibi) [reply bun.lockb:1]: It's similar to package-lock.json but faster to read and write, and more compact due to its binary format. It was accidentally pushed, so we removed it as well in  [#cdd7bb2](https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/79/commits/cdd7bb235e2ff7bd0b61ec02abc5089d31a9eef1)

File: app/topics/page.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        description:
+            "Simplifying bitcoin tech to help you learn as efficiently as possible"
+    }
+})
+
+// Default page to display in home page for topics
+const slug = "welcome"
+
+export default async function Page() {
+    // Filter out drafts in production
```
[the remark this answers is not in the document]
Comment (jrakibi) [reply app/topics/page.tsx]: Extracted this into the getTopicData() function in [commit #b92da70](https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/79/commits/b92da70fbc6bf4c9bc860d3913d2b3d5b2e1f2c7)

**Review by jrakibi — COMMENTED** [review]
Thanks for the review, @Extheoisah 

To simplify the process, I addressed each major feedback in its own commit. I hope this makes reviewing easier

I also addressed all the nits and minor updates in [commit #3413095](https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/79/commits/341309548fe02ddfd8ee0bf8f017d09022b80839) , so I haven't left individual comment replies on them

**Comment by Extheoisah** [comment]
On mobile this long script causes an overflow
Screenshot_20240809_085243_Samsung Internet


And the script stack visualizer looks off on mobile. Although not sure what we could do about that.
Screenshot_20240809_085324_Samsung Internet

**Comment by jrakibi** [comment]
@Extheoisah This [commit](https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/79/commits/453276123e8c10f31414b39c387cbd46bacf6bc9) fixes the responsiveness issue for the script visualizer on mobile

**Comment by Extheoisah** [comment]
@jrakibi In dark mode, some text are not legible
image

image

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**